pkg/cableengine/cableengine.go

```diff
@@ -145,37 +145,35 @@ func (i *engine) installCableWithNATInfo(rnat *natdiscovery.NATEndpointInfo) err
 	for _, active := range activeConnections {
 		klog.V(log.TRACE).Infof("Analyzing currently active connection %q", active.Endpoint.CableName)
 
-		disconnect := false
+		if active.Endpoint.ClusterID != endpoint.Spec.ClusterID {
+			continue
+		}
+
+		klog.V(log.DEBUG).Infof("Found a pre-existing cable %q that belongs to this cluster %s",
+			active.Endpoint.CableName, endpoint.Spec.ClusterID)
 
-		if active.Endpoint.CableName == endpoint.Spec.CableName {
-			// There could be scenarios where the cableName would be the same but the
-			// PublicIP of the active GatewayNode changes.
+		prevTimestamp := i.installedCables[active.Endpoint.CableName]
+		if endpoint.CreationTimestamp.Before(&prevTimestamp) {
+			klog.Warningf("The timestamp (%s) for new cable %q is older than the timestamp (%s) of the pre-existing "+
+				"cable %q - not replacing", endpoint.CreationTimestamp, endpoint.Spec.CableName, prevTimestamp, active.Endpoint.CableName)
+			return nil
+		}
+
+		if endpoint.CreationTimestamp.Equal(&prevTimestamp) && active.Endpoint.CableName == endpoint.Spec.CableName {
+			// There could be scenarios where the cableName would be the same but the PublicIP of the active GatewayNode changes.
 			if active.UsingIP == rnat.UseIP && active.UsingNAT == rnat.UseNAT {
```

Comment:
Should we add also the compare for the BackendConfig? , I guess if it's going to be re-created it actually doesn't matter. As long as we don't change that behavior in admiral. But if we check the backend config it would allow us to change that behavior in the future if we wanted I guess.

Anyway, just a thought.

Reply:
yeah I was going to do that in a separate PR but I can do it with this one.

Reply:
Done